Sources/Sentry/SentryUIPerformanceTracker.m

```diff
@@ -0,0 +1,308 @@
+#import "SentryUIPerformanceTracker.h"
+#import "SentryHub.h"
+#import "SentryLog.h"
+#import "SentryPerformanceTracker.h"
+#import "SentrySDK+Private.h"
+#import "SentryScope.h"
+#import "SentrySpanId.h"
+#import "SentrySwizzle.h"
+#import "UIViewControllerHelper.h"
+#import <objc/runtime.h>
+
+#if SENTRY_HAS_UIKIT
+#    import <UIKit/UIKit.h>
+#endif
+
+static NSString *const SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID
+    = @"SENTRY_UI_PERFORMANCE_TRACKER_SPAN_ID";
+
+static NSString *const SENTRY_UI_PERFORMANCE_TRACKER_LAYOUTSUBVIEW_SPAN_ID
+    = @"SENTRY_UI_PERFORMANCE_TRACKER_LAYOUTSUBVIEW_SPAN_ID";
+
+/**
+ *Auxiliary class to store UIViewController performance monitoring spanid.
+ */
+@interface SentryViewControllerPerformanceSpans : NSObject
+
+@property (nonatomic, strong) SentrySpanId *mainSpan;
+
+@property (nonatomic, strong) SentrySpanId *layoutSubViewsSpan;
+
+- (instancetype)initWithMainSpan:(SentrySpanId *)mainSpan;
+
+@end
+
+@implementation SentryViewControllerPerformanceSpans
+
+- (instancetype)initWithMainSpan:(SentrySpanId *)mainSpan
+{
+    if (self = [super init]) {
+        self.mainSpan = mainSpan;
+    }
+    return self;
+}
+
+@end
+
+@interface
+SentryUIPerformanceTracker ()
++ (NSMutableDictionary *)swizzeledViewControllers;
+@end
+
+@implementation SentryUIPerformanceTracker
+
++ (void)start
+{
+// If there`s no UIKIT don`t need to try the swizzling.
+#if SENTRY_HAS_UIKIT
+    [SentryUIPerformanceTracker swizzleViewControllerInits];
+#else
+    [SentryLog logWithMessage:@"NO UIKit -> [SentryUIPerformanceTracker "
+                              @"start] does nothing."
+                     andLevel:kSentryLevelDebug];
+#endif
+}
+
+/**
+ *Global dictionary to store monitored UIViewControllers span ids.
+ */
++ (NSMutableDictionary *)swizzeledViewControllers
+{
+    static NSMutableDictionary *_swizzeledViewControllers;
+    if (_swizzeledViewControllers == nil)
+        _swizzeledViewControllers = [[NSMutableDictionary alloc] init];
+
+    return _swizzeledViewControllers;
+}
+
+/**
+ * Swizzle the some init methods of the view controller,
+ * so we can swizzle user view controller subclass on demand.
```

Comment:
`l`: Can we maybe add a comment on why we swizzle subclasses? Because we want to measure the time of the method of the subclass and not the base class.